### PR TITLE
attitude_estimator_q Run() method refactoring

### PR DESCRIPTION
**Pull request by mcsauder**
**Describe problem solved by this pull request**
This PR breaks up the `attitude_estimator_q_main.cpp`  `Run()` method into a sequence granular functions to better encapsulate what each method is accomplishing.

**Describe your solution**
This PR is straightforward refactoring, no functional logic has been changed in the class.

**Test data / coverage**
Flight logs to follow when weather is more conducive.

**Additional Context**
Due to ugly nature of the single diff, I've broken this refactoring into a series of commits that are relatively straightforward to inspect.

If you have any questions or feedback on this PR please let me know!  Thanks!

-Mark

### CODE REVIEW

File: src/modules/attitude_estimator_q/attitude_estimator_q_main.cpp [hunk 1]
```diff
@@ -109,13 +123,17 @@ class AttitudeEstimatorQ : public ModuleBase<AttitudeEstimatorQ>, public ModuleP
...
+	uORB::Subscription 		_vehicle_attitude_sub{ORB_ID(vehicle_attitude)};
+	uORB::Subscription 		_vehicle_gps_position_sub{ORB_ID(vehicle_gps_position)};
+	uORB::Subscription 		_vehicle_local_position_sub{ORB_ID(vehicle_local_position)};
+	uORB::Subscription 		_vehicle_magnetometer_sub{ORB_ID(vehicle_magnetometer)};
+	uORB::Subscription 		_vehicle_mocap_odometry_sub{ORB_ID(vehicle_mocap_odometry)};
+	uORB::Subscription 		_vehicle_visual_odometry_sub{ORB_ID(vehicle_visual_odometry)};
+
+	uORB::Publication<vehicle_attitude_s> _vehicle_attitude_pub{ORB_ID(vehicle_attitude)};
+
+	sensor_combined_s _sensors;
```
[the remark this answers is not in the document]
Comment (mcsauder) [reply src/modules/attitude_estimator_q/attitude_estimator_q_main.cpp]: Excellent advice!  Fixed!

**Comment by mcsauder** [comment]
Rebased on master and re-flown with attitude_estimator_q as the primary attitude estimate: https://review.px4.io/plot_app?log=0971bda8-b6ef-4585-ae1e-8799aadb7e45

**Comment by mcsauder** [comment]
Rebased against current master.

**Comment by mcsauder** [comment]
Commits squashed and rebased against current master.

File: src/modules/attitude_estimator_q/attitude_estimator_q_main.cpp [hunk 2]
```diff
@@ -220,154 +299,99 @@ AttitudeEstimatorQ::Run()
...
-void
-AttitudeEstimatorQ::update_parameters(bool force)
+void AttitudeEstimatorQ::update_parameters(bool force)
 {
 	// check for parameter updates
 	if (_parameter_update_sub.updated() || force) {
 		// clear update
 		parameter_update_s pupdate;
-		_parameter_update_sub.copy(&pupdate);
+		_parameter_update_sub.update(&pupdate);
```
Comment (dagar) [inline src/modules/attitude_estimator_q/attitude_estimator_q_main.cpp:394]: I would keep it as a forced copy() (consistent with the other usage) and easier to refactor out later on if we come up with something better.

**Comment by dagar** [comment]
Seems fine to me if you've tested this thoroughly (we don't have automated testing this module).

**Comment by mcsauder** [comment]
> Seems fine to me if you've tested this thoroughly (we don't have automated testing this module).

We've been flying this refactor for the past few weeks and reviewing logs.  Flight data with/without this refactor behaves the same.

**Review by dagar — APPROVED** [review]

**Comment by mcsauder** [comment]
Thanks @dagar !

**Pull request merged**